src/utils/canvas/getCanvasBoundingBox.ts

```diff
@@ -25,29 +50,69 @@ function checkColumn(data: Uint8ClampedArray, width: number, x: number, top: num
 }
 
 /**
- * Measuring the bounds of a canvas' visible (non-transparent) pixels.
- * @param canvas - The canvas to measure.
- * @param resolution - The resolution of the canvas.
- * @returns The bounding box of the canvas' visible pixels.
+ * Measures the bounding box of a canvas's visible (non-transparent) pixels.
+ *
+ * This function analyzes the alpha channel of the canvas pixels to find the smallest
+ * rectangle containing all non-transparent pixels. It's useful for optimizing sprite
+ * rendering by trimming transparent borders.
+ *
+ * Uses an internal canvas with `willReadFrequently: true` for efficient pixel data access.
+ * This internal canvas is reused between calls for better performance.
+ * @example
+ * ```typescript
+ * // Basic usage - get trim bounds at full resolution
+ * const bounds = getCanvasBoundingBox(myCanvas);
+ * console.log(bounds); // Rectangle{x: 10, y: 5, width: 100, height: 200}
+ * // Optimized for performance with lower resolution scanning
+ * const fastBounds = getCanvasBoundingBox(largeCanvas, canvas.width, canvas.height, 0.5);
+ * // Resolution of 0.5 means scanning at half size, much faster for large canvases
+ *
+ * // Using custom dimensions - only analyze part of the canvas
+ * const partialBounds = getCanvasBoundingBox(myCanvas, 100, 100);
+ * // Only analyzes a 100x100 region starting from top-left
+ * ```
+ * @param canvas - The canvas to measure
+ * @param width - Optional. The width to analyze (defaults to canvas.width)
+ * @param height - Optional. The height to analyze (defaults to canvas.height)
+ * @param resolution - Optional. The resolution at which to analyze the canvas, between 0-1.
+ *                    Lower values improve performance for large canvases but may be less precise.
+ *                    Default is 1 (full resolution).
+ * @param output - Optional. The rectangle to store the result in.
+ * @returns The bounding box as a Rectangle containing the visible content.
+ *          Returns Rectangle.EMPTY if the canvas is completely transparent.
  * @since 7.1.0
  * @memberof utils
  */
-export function getCanvasBoundingBox(canvas: ICanvas, resolution = 1): Rectangle
+export function getCanvasBoundingBox(
+    canvas: ICanvas,
+    width: number = canvas.width,
+    height: number = canvas.height,
+    resolution = 1,
```

Comment:
We should keep resolution as the second option for backward compatibility 
or convert this to an object now we have many options

Reply:
added deprication!